astropy/io/fits/diff.py

```diff
@@ -149,21 +150,45 @@ def report(self, fileobj=None, indent=0, clobber=False):
             The number of 4 space tabs to indent the report.
 
         clobber : bool
-            Whether the report output should overwrite an existing file, when
-            fileobj is specified as a path.
+            If ``True`` and fileobj is a path, overwrite the output
+            file if it exist.
+            The use of clobber is deprecated; use overwrite instead.
+
+        overwrite : bool
+            If ``True`` and fileobj is a path, overwrite the output
+            file if it exist
+
+            If both clobber and overwrite are set to a boolean value
+            (whether equal or not), an exception is raised. If clobber
+            is used instead of overwrite, an AstropyDeprecationWarning
+            is issued.
 
         Returns
         -------
         report : str or None
+
         """
 
         return_string = False
         filepath = None
 
+        if isinstance(clobber, bool) and isinstance(overwrite, bool):
```

Comment:
It seems like `clobber` just had to be "truthy" but not necessarily be a boolean. 

Say for example I set `clobber=1, overwrite=False` then this wouldn't throw the error even though I specified both of them.